sbol_utilities/sbol3_sbol2_conversion.py

```diff
@@ -584,26 +596,28 @@ def visit_module(self, a: sbol2.Module):
         raise NotImplementedError('Conversion of Module from SBOL2 to SBOL3 not yet implemented')
 
     def visit_module_definition(self, md: sbol2.ModuleDefinition):
-        # Make the Component object and add it to the document
-        c3 = sbol3.Component(self._sbol3_identity(md), types=md.type, roles=md.roles, namespace=self._sbol3_namespace(md))
+        # ModuleDefinitions convert to SBOL3 Components annotated as "functional entities"
+        c3 = sbol3.Component(self._sbol3_identity(md), types=[sbol3.SBO_FUNCTIONAL_ENTITY], roles=md.roles, namespace=self._sbol3_namespace(md))
 
         for i2 in md.interactions:
             i3 = self.visit_interaction(i2)
             c3.interactions.append(i3)
             self.update_identity(i2, i3)
 
-        c3.interface = sbol3.Interface()
-        for fc in md.functionalComponents:
-            sc = self.visit_functional_component(fc)
-            c3.features.append(sc)
-            self.update_identity(fc, sc)
-            if fc.direction == 'http://sbols.org/v2#in' or fc.direction == 'http://sbols.org/v2#inout':
-                c3.interface.inputs.append(sc)
-            if fc.direction == 'http://sbols.org/v2#out' or fc.direction == 'http://sbols.org/v2#inout':
-                c3.interface.outputs.append(sc)
-            if fc.direction == 'http://sbols.org/v2#none':
-                c3.interface.nondirectionals.append(sc)
+        if md.functionalComponents:
+            c3.interface = sbol3.Interface()
```

Comment:
Interface shouldn't be unconditionally created - it should only be added if there is at least one FunctionalComponent with a direction other than `none`.

Reply:
Nice catch!

Reply:
On second thought, before making any further changes here, we need to clarify when it is appropriate to use the `Interface.nondirectionals` property.  Two issues:

1. If a `FunctionalComponent` has no direction,  in what cases, if any, would it be appropriate to add it to an `Interface` as a nondirectional `Feature`?
2. If a `FunctionalComponent` has both in and out directionality, would we add it to an `Interface` both as an input and output, or do we add it as a nondirectional `Feature`?